### PR TITLE
Allow using explicit colormapping on non-categorical data

### PR DESCRIPTION
**Pull request by philippjfr**
This PR allows providing an explicit colormapping for non-categorical data.

- [x] Closes https://github.com/ioam/holoviews/issues/3033
- [x] Adds unit test

### CODE REVIEW

File: holoviews/plotting/bokeh/element.py [hunk 1]
```diff
@@ -1189,6 +1192,11 @@ def _get_color_data(self, element, ranges, style, name='color', factors=None, co
 
         mapper = self._get_colormapper(cdim, element, ranges, style,
                                        factors, colors)
+        if not factors and isinstance(mapper, CategoricalColorMapper):
+            field += '_str'
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/element.py]: Looks a little hackish?

Comment (philippjfr) [reply holoviews/plotting/bokeh/element.py]: Does it? In what way?

Comment (philippjfr) [reply holoviews/plotting/bokeh/element.py]: To be clearer, what's your suggestion for not overriding the original column and making it clear the column has been "categorized" by casting values to strings?

**Comment by jlstevens** [comment]
Looks good! I'll merge when the tests pass.

**Comment by jlstevens** [comment]
Tests have passed now. Merging.

**Pull request merged**